src/command/process_update_task.rs

```diff
@@ -344,6 +508,68 @@ impl UpdateProcessor {
     }
 }
 
+#[derive(Serialize, Deserialize, Debug, TypedBuilder, Eq, PartialEq, Clone)]
+#[serde(crate = "fang::serde")]
+pub struct ScheduleWeatherTask {
+    user_hour: i8,
+    minutes: i8,
+    offset: i8,
+    username: String,
+    chat_id: i64,
+    default_city_id: i32,
+}
+
+#[typetag::serde]
+#[async_trait]
+impl AsyncRunnable for ScheduleWeatherTask {
+    async fn run(&self, _queueable: &mut dyn AsyncQueueable) -> Result<(), FangError> {
+        // here we should program the weather_info deliver
+        let repo = Repo::repo().await.unwrap();
+        let api = ApiClient::api_client().await;
+
+        let city = repo.search_city_by_id(&self.default_city_id).await.unwrap();
+
+        let weather_client = WeatherApiClient::weather_client().await;
+
+        let weather_info = weather_client
+            .fetch_weekly(city.coord.lat, city.coord.lon)
+            .await
+            .unwrap();
+
+        let text = format!(
+            "Hi {} !, this is your scheduled weather info.\n\n {},{}\nLat {} , Lon {}\n{}",
+            self.username, city.name, city.country, city.coord.lat, city.coord.lon, weather_info,
+        );
+
+        api.send_message_without_reply(self.chat_id, text)
+            .await
+            .unwrap();
+
+        Ok(())
+    }
+
+    fn uniq(&self) -> bool {
+        true
+    }
+
+    fn task_type(&self) -> String {
+        TASK_TYPE.to_string()
+    }
+
+    fn cron(&self) -> Option<Scheduled> {
+        let hour_utc = if self.user_hour - self.offset < 0 {
+            self.user_hour - self.offset + 24
```

Comment:
I think chrono supports timezone conversions

https://github.com/ayrat555/el_monitorro/blob/master/src/deliver/render_message.rs#L115

Reply:
That is not possible in my case because we do not have a date to work because date it is generated inside fang with the cron expression. 

So here we are converting a cron expression into UTC not only a date.

Reply:
We are converting a entire range of DateTime not only one so this chrono func wont work.